src/NationalInstruments.Analyzers/Style/ChainOfMethodsWithLambdasAnalyzer.cs

```diff
@@ -36,25 +36,52 @@ public override void Initialize(AnalysisContext context)
             context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
 
             context.RegisterSyntaxNodeAction(
-                AnalyzeSyntaxNode,
+                AnalyzeSyntaxContext,
                 SyntaxKind.ExpressionStatement,
                 SyntaxKind.EqualsValueClause,
                 SyntaxKind.Argument,
                 SyntaxKind.ArrowExpressionClause);
+
+            context.RegisterSyntaxNodeAction(
+                AnalyzeArrayInitializerContext,
+                SyntaxKind.ArrayInitializerExpression);
         }
 
-        private static void AnalyzeSyntaxNode(SyntaxNodeAnalysisContext context)
+        private static void AnalyzeSyntaxContext(SyntaxNodeAnalysisContext context)
         {
             var parentSyntaxNode = context.Node;
 
             // Find the invocation expression i.e
             // a method/delegate call or a property access
             // or a chain of them
             var invocationExpressionSyntax = parentSyntaxNode
-                .DescendantNodes()
+                .DescendantNodes(IsNotArrayCreationSyntax)
                 .OfType<InvocationExpressionSyntax>()
                 .FirstOrDefault();
 
+            AnalyzeSyntaxNode(invocationExpressionSyntax, context.ReportDiagnostic);
+
+            bool IsNotArrayCreationSyntax(SyntaxNode syntaxNode) => !syntaxNode.IsKind(SyntaxKind.ArrayCreationExpression);
+        }
+
+        private static void AnalyzeArrayInitializerContext(SyntaxNodeAnalysisContext context)
+        {
+            var arrayInitializerSyntaxNode = context.Node;
+
+            // Find only direct child invocation expressions
+            var invocationExpressions = arrayInitializerSyntaxNode
```

Comment:
`invocationExpressionSyntaxes`. My convention is to suffix anything that's syntax with "Syntax" so if I get the symbol I can just call it the same thing w/o "Syntax"

Reply:
Fixed [here](https://github.com/ni/csharp-styleguide/pull/65/commits/d6f8e929e94e3ba597336ec3a1a1108980eecd1d).